packages/graphql-language-service/src/interface/getDiagnostics.ts

```diff
@@ -81,15 +81,22 @@ export function getDiagnostics(
   try {
     ast = parse(query);
   } catch (error) {
-    const range = getRange(error.locations[0], query);
-    return [
-      {
-        severity: DIAGNOSTIC_SEVERITY.Error as DiagnosticSeverity,
-        message: error.message,
-        source: 'GraphQL: Syntax',
-        range,
-      },
-    ];
+    if (error instanceof GraphQLError) {
+      const range = getRange(
+        error.locations?.[0] ?? { line: 0, column: 0 },
+        query,
+      );
+
+      return [
+        {
+          severity: DIAGNOSTIC_SEVERITY.Error as DiagnosticSeverity,
+          message: error.message,
+          source: 'GraphQL: Syntax',
+          range,
+        },
+      ];
+    }
+    throw error;
```

Comment:
Please do not throw this error. we trap  and report them as needed in language services, we don't throw. This would crash graphiql or the LSP server on syntax errors

Reply:
ohh right ok. in that case, I would prefer it log the message but still trap the exception